src/cli/export/export.go

```diff
@@ -37,3 +55,119 @@ func exportCmd() *cobra.Command {
 func handleExportCmd(cmd *cobra.Command, args []string) error {
 	return cmd.Help()
 }
+
+func runExport(
+	ctx context.Context,
+	cmd *cobra.Command,
+	args []string,
+	ueco utils.ExportCfgOpts,
+	sel selectors.Selector,
+	backupID, serviceName string,
+) error {
+	r, _, _, _, err := utils.GetAccountAndConnect(ctx, sel.PathService(), repo.S3Overrides(cmd))
+	if err != nil {
+		return Only(ctx, err)
+	}
+
+	defer utils.CloseRepo(ctx, r)
+
+	exportLocation := args[0]
+	if exportLocation == "" {
+		// This should not be possible, but adding it just in case.
+		exportLocation = control.DefaultRestoreLocation + dttm.FormatNow(dttm.HumanReadableDriveItem)
+	}
+
+	Infof(ctx, "Exporting to folder %s", exportLocation)
+
+	eo, err := r.NewExport(
+		ctx,
+		backupID,
+		sel,
+		utils.MakeExportConfig(ctx, ueco),
+	)
+	if err != nil {
+		return Only(ctx, clues.Wrap(err, "Failed to initialize "+serviceName+" export"))
+	}
+
+	expColl, err := eo.Run(ctx)
+	if err != nil {
+		if errors.Is(err, data.ErrNotFound) {
+			return Only(ctx, clues.New("Backup or backup details missing for id "+backupID))
+		}
+
+		return Only(ctx, clues.Wrap(err, "Failed to run "+serviceName+" export"))
+	}
+
+	// It would be better to give a progressbar than a spinner, but we
+	// have know way of knowing how many files are available as of now.
+	diskWriteComplete := observe.MessageWithCompletion(ctx, "Writing data to disk")
+	defer func() {
+		diskWriteComplete <- struct{}{}
+		close(diskWriteComplete)
+	}()
+
+	err = writeExportCollections(ctx, exportLocation, expColl)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func writeExportCollections(
+	ctx context.Context,
+	exportLocation string,
+	expColl []export.Collection,
+) error {
+	for _, col := range expColl {
+		folder := filepath.Join(exportLocation, col.BasePath())
+
+		for item := range col.Items(ctx) {
+			err := item.Error
+			if err != nil {
+				return Only(ctx, clues.Wrap(err, "getting item").With("dir_name", folder))
+			}
+
+			err = writeExportItem(ctx, item, folder)
+			if err != nil {
+				return err
+			}
+		}
+	}
+
+	return nil
+}
+
+// writeExportItem writes an ExportItem to disk in the specified folder.
+func writeExportItem(ctx context.Context, item export.Item, folder string) error {
+	name := item.Data.Name
+	fpath := filepath.Join(folder, name)
+
+	progReader, pclose := observe.ItemSpinner(
+		ctx,
+		item.Data.Body,
+		observe.ItemExportMsg,
+		clues.Hide(name))
+
+	defer item.Data.Body.Close()
+	defer pclose()
+
+	err := os.MkdirAll(folder, os.ModePerm)
+	if err != nil {
+		return Only(ctx, clues.Wrap(err, "creating directory").With("dir_name", folder))
```

Comment:
please add `.WithCtx(ctx)` to all of these errors, since ctx isn't passed into the called func.
```suggestion
		return Only(ctx, clues.Wrap(err, "creating directory").With("dir_name", folder).WithClues(ctx))
```